cstar/tests/config.py

```diff
@@ -0,0 +1,70 @@
+"""
+Configuration module for testing C-Star with pytest.
+
+The first section sets or gets any globally accessible paths (e.g. that of the test directory itself, TEST_DIRECTORY).
+
+The second section defines the `TEST_CONFIG` dictionary, whose keys serve as parameters for the primary test in
+cstar/tests/roms/test_cstar_test_blueprints.py.
+The entries in this dictionary should specify a path to a template blueprint file to be modified, and a dictionary
+of substrings in that file to modify (e.g. {"oldstr" : "newstr"}). These will be combined to make a modified blueprint
+from which the test case is constructed.
+"""
+
+from pathlib import Path
+from pooch import os_cache
+
+## Common paths used in tests
+
+
+def _get_test_directory():
+    from importlib.util import find_spec
+    from pathlib import Path
+
+    spec = find_spec("cstar")
+    if spec is not None:
+        return Path(spec.origin).parent / "tests"
+    else:
+        raise RuntimeError("Cannot determine the package location of cstar")
+
+
+TEST_DIRECTORY = _get_test_directory()
+ROMS_TOOLS_DATA_DIRECTORY = Path(os_cache("roms_tools_datasets_for_cstar_test_case"))
+CSTAR_TEST_DATA_DIRECTORY = Path(os_cache("cstar_test_case_data"))
```

Comment:
Why can't these also be temporary directories, provided via fixtures to the other fixtures (e.g. `fetch_roms_tools_source_data`)?

Reply:
I've borrowed the roms-tools implementation where external data that are _needed by but not created by_ the test suite is fetched to a cache dir - a little softer than a temporary directory but allows re-use locally between tests without affecting the test outcomes themselves.

Reply:
to the second part, having a global name in the config module allows us to define where they go in a central location that is used consistently elsewhere. The directories aren't actually created here (I don't think), the path is just defined. They're then created in fixtures which use these vars.

Reply:
> a cache dir - a little softer than a temporary directory but allows re-use locally between tests without affecting the test outcomes themselves.

But isn't it possible to achieve the same thing using a pytest `tmpdir` fixture with `scope='session'` or similar?